### ISSUE TITLE
R5 Network loading from directory must give some order to files to avoid discrepancy in link ids 

### ISSUE THREAD

**Issue by REASY** [LBNL-UCB-STI/beam#1743]
[TransportNetwork fromDirectory](
https://github.com/conveyal/r5/blob/dev/src/main/java/com/conveyal/r5/transit/TransportNetwork.java#L188) uses [File.listFiles()](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#listFiles()) which is:

> There is no guarantee that the name strings in the resulting array will appear in any specific order; they are not, in particular, guaranteed to appear in alphabetical order.

So we have to make sure that there is some order in filenames to be able to use link ids across different operating systems/JVMs and use [TransportNetwork fromFiles](https://github.com/conveyal/r5/blob/dev/src/main/java/com/conveyal/r5/transit/TransportNetwork.java#L178)

**Comment by michaz** [comment]
Is this something we are actually observing -- different people getting different link ids on the same scenario -- or just a possibility?

If we are actually observing it, it remains possible that there are more places in R5 that produce non-stable ids.

Can't we just say "don't do that, link IDs _are_ unstable"?

**Comment by REASY** [comment]
@michaz, yes, we have faced it. In case of cluster mode it might be real issue for us. Also during debugging of travel times I had to every time copy `network.dat` and `physSimNetwork.xml` to my machine from the machines where run happened to make sure that I see _the same_ links. So I think it worth to fix it. For me it looks pretty straightforward, we just need to give some order to the files and it ids will be stable.
